client/views/account/accountProfile.coffee

```diff
@@ -38,7 +46,9 @@ Template.accountProfile.onCreated ->
 
 	@save = ->
 		instance = @
-
+		if !RocketChat.settings.get("Accounts_AllowUsernameChange")
```

Comment:
This needs to check if there is a value in the field before just simply returning from here. As this prevents the email, password, language, etc from being changed.